server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/TermsAggregationBuilder.java

```diff
@@ -135,7 +136,22 @@ public boolean supportsSampling() {
     }
 
     @Override
-    public boolean supportsParallelCollection() {
+    public boolean supportsParallelCollection(ToLongFunction<String> fieldCardinalityResolver) {
+        // TODO unconditionally enable parallelism when ordered by key instead of count as precision is not affected
+        /*
+        if (InternalOrder.isKeyOrder(order)) {
+            return true;
+        }
+        */
+        if (script() == null
+            && (executionHint == null || executionHint.equals(TermsAggregatorFactory.ExecutionMode.GLOBAL_ORDINALS.toString()))) {
+            long cardinality = fieldCardinalityResolver.applyAsLong(name);
+            if (cardinality <= 0) {
```

Comment:
0 is used in some places for unmapped fields, I think. While -1 means we don't know the cardinality, hence we should stay on the safe side and not parallelize, does it make any sense to parallelize when cardinality is 0?

Reply:
> does it make any sense to parallelize when cardinality is 0?

I think we should also parallelize when cardinality is zero. It is likely there there are other aggregations defined too? And then we can't run in parallel, because a terms agg is defined with a field that is empty.
Additionally if the `missing` parameter is used, then we still have a bucket and potential sub aggs need to be executed.

Reply:
sounds good to me.